appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/DesignToolbar.java

```diff
@@ -159,145 +136,44 @@ public enum View {
   // on the device.
   public static LinkedList<String> pushedScreens = Lists.newLinkedList();
 
-  // Is the Gallery Enabled (new gallery)?
-  private boolean galleryEnabled = false;
+  interface DesignToolbarUiBinder extends UiBinder<Toolbar, DesignToolbar> {}
+  private static final DesignToolbarUiBinder UI_BINDER = GWT.create(DesignToolbarUiBinder.class);
+
+  @UiField DropDownButton pickFormItem;
+  @UiField ToolbarItem addFormItem;
+  @UiField ToolbarItem removeFormItem;
+  @UiField ToolbarItem switchToDesign;
+  @UiField ToolbarItem switchToBlocks;
 
   /**
    * Initializes and assembles all commands into buttons in the toolbar.
    */
   public DesignToolbar() {
     super();
 
-    isReadOnly = Ode.getInstance().isReadOnly();
-    galleryEnabled = Ode.getInstance().getSystemConfig().getGalleryEnabled();
-    projectNameLabel = new Label();
-    projectNameLabel.setStyleName("ya-ProjectName");
-    HorizontalPanel toolbar = (HorizontalPanel) getWidget();
-    toolbar.insert(projectNameLabel, 0);
-
-    // width of palette minus cellspacing/border of buttons
-    toolbar.setCellWidth(projectNameLabel, "222px");
-
-    addButton(new ToolbarItem(WIDGET_NAME_TUTORIAL_TOGGLE,
-        MESSAGES.toggleTutorialButton(), new ToogleTutorialAction()));
-    setButtonVisible(WIDGET_NAME_TUTORIAL_TOGGLE, false); // Don't show unless needed
-
-    List<DropDownItem> screenItems = Lists.newArrayList();
-    addDropDownButton(WIDGET_NAME_SCREENS_DROPDOWN, MESSAGES.screensButton(), screenItems);
-
-    if (AppInventorFeatures.allowMultiScreenApplications() && !isReadOnly) {
-      addButton(new ToolbarItem(WIDGET_NAME_ADDFORM, MESSAGES.addFormButton(),
-          new AddFormAction()));
-      addButton(new ToolbarItem(WIDGET_NAME_REMOVEFORM, MESSAGES.removeFormButton(),
-          new RemoveFormAction()));
-    }
-    if (galleryEnabled && !Ode.getInstance().getGalleryReadOnly()) {
-      addButton(new ToolbarItem(WIDGET_NAME_SENDTOGALLERY,
-          MESSAGES.publishToGalleryButton(), new SendToGalleryAction()));
+    populateToolbar(UI_BINDER.createAndBindUi(this));
+    if (Ode.getInstance().isReadOnly() || !AppInventorFeatures.allowMultiScreenApplications()) {
+      removeItem(addFormItem);
+      removeItem(removeFormItem);
```

Comment:
Note that this didn't seem to work. If I go to ai2-test as an admin and masquerade as another user, I still get the add/remove screen buttons.